From: SeongJae Park <sj@kernel.org>
Date: Sat, 26 Jul 2025 10:40:46 -0700
Subject: [PATCH] mm/damon/core: remove debugging purpose nr_accesses
 corruption check

It is now being checked, warned, and fixed by
damon_warn_fix_nr_accesses_corruption().  Remove the debugging purpose
code.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 9a801c1faa8e..7fbb00884198 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2379,14 +2379,6 @@ static void damon_merge_regions_of(struct damon_target *t, unsigned int thres,
 		else
 			r->age++;
 
-		if (r->nr_accesses != r->nr_accesses_bp / 10000) {
-			pr_err("nr_accesses (%u) != nr_accesses_bp (%u)\n",
-					r->nr_accesses, r->nr_accesses_bp);
-#if DAMON_PADDR_FAULT_TESTING == 0
-			BUG();
-#endif
-		}
-
 		if (prev && prev->ar.end == r->ar.start &&
 		    abs(prev->nr_accesses - r->nr_accesses) <= thres &&
 		    damon_sz_region(prev) + damon_sz_region(r) <= sz_limit)
-- 
2.39.5

